crates/extensions/tedge_config_manager/src/actor.rs

```diff
@@ -294,12 +294,17 @@ impl ConfigManagerActor {
         // move to destination later
         let temp_path = &self.config.tmp_path.join(&file_entry.config_type);
 
-        let download_request = DownloadRequest::new(&request.tedge_url, temp_path.as_std_path())
+        let Some(tedge_url) = &request.tedge_url else {
+            debug!("tedge_url not present in config update payload, ignoring");
+            return Ok(());
```

Comment:
I guess it's too late now, but it feels like properly using the scheduled state (the agent updating the state from `init` to `schedule` after the download, and then this config manager actor reacting only to that `scheduled` state) would have been clearer, in terms of the operation control flow, IMO. Though it works, reacting to the `executing` state differently at different times looks a bit convoluted.

Reply:
But arguably, downloading the config the file is a normal part of execution of the `config_file` operation, and most of the actual time of the operation execution is spent there, so I don't think downloading in `init` or `schedule` state would be better. For other complex operations there can be multiple steps for which we don't have enough distinct states. I do agree, however, that it's a bit hard to track and I expect that perhaps custom workflows and some other refactorings I plan for operation code would at least alleviate this somewhat.

Reply:
Yeah, indeed too late to fix that, but it would have been better to introduce a specific state - say downloading.

>  I do agree, however, that it's a bit hard to track and I expect that perhaps custom workflows and some other refactorings I plan for operation code would at least alleviate this somewhat.

I expect the same ;-)